apps/auth0-rules/src/add-user-metadata.ts

```diff
@@ -1,10 +1,63 @@
 import type { User } from '@asap-hub/auth';
-import type { UserMetadataResponse } from '@asap-hub/model';
+import type { gp2, UserMetadataResponse, UserResponse } from '@asap-hub/model';
 import got from 'got';
 import { URL, URLSearchParams } from 'url';
 import { handleError } from './handle-error';
 import type { Rule } from './types';
 
+type Auth0UserResponse = UserMetadataResponse | gp2.UserResponse;
+
+function isUserResponse(
+  response: UserMetadataResponse | gp2.UserResponse,
+): response is UserResponse & { algoliaApiKey: string } {
+  return 'onboarded' in (response as UserMetadataResponse);
+}
+
+const extractUser = (response: Auth0UserResponse) => {
+  if (isUserResponse(response)) {
+    const {
+      id,
+      onboarded,
+      displayName,
+      email,
+      firstName,
+      lastName,
+      avatarUrl,
+      teams,
+      algoliaApiKey,
+    } = response;
+
+    return {
+      id,
+      onboarded,
+      displayName,
+      email,
+      firstName,
+      lastName,
+      avatarUrl,
+      teams: teams.map((team) => ({
+        id: team.id,
+        displayName: team.displayName,
+        role: team.role,
+      })),
+      algoliaApiKey,
+    };
+  }
+  const { id, email, firstName, lastName, avatarUrl } = response;
+
+  return {
+    id,
+    displayName: '',
```

Comment:
actually i think we have displayName in the gp2 user response?

Reply:
done